pyproject.toml

```diff
@@ -1,3 +1,122 @@
+[project]
+name = "astropy"
+dynamic = [
+    "version"
+]
+description = "Astronomy and astrophysics core library"
+readme = "README.rst"
+authors = [
+    { name = 'The Astropy Developers', email = 'astropy.team@gmail.com' }
+]
+license = { file = "LICENSE.rst" }
+requires-python = ">=3.9"
+classifiers = [
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: OS Independent",
+    "Programming Language :: C",
+    "Programming Language :: Cython",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Topic :: Scientific/Engineering :: Astronomy",
+    "Topic :: Scientific/Engineering :: Physics",
+]
+keywords = [
+    "astronomy",
+    "astrophysics",
+    "cosmology",
+    "space",
+    "science",
+    "units",
+    "table",
+    "wcs",
+    "samp",
+    "coordinate",
+    "fits",
+    "modeling",
+    "models",
+    "fitting",
+    "ascii",
+]
+dependencies = [
+    "numpy>=1.22",
+    "pyerfa>=2.0",
+    "astropy-iers-data>=0.2023.7.31.0.32.45",
+    "PyYAML>=3.13",
+    "packaging>=19.0",
+]
+
+[project.optional-dependencies]
+test = [
+    "pytest>=7.0",
+    "pytest-doctestplus>=0.12",
+    "pytest-astropy-header>=0.2.1",
+    "pytest-astropy>=0.10",
+    "pytest-xdist",
+]
+test_all = [
+    "astropy[test]",  # installs the [test] dependencies
+    "objgraph",
+    "ipython>=4.2",
+    "coverage[toml]",
+    "skyfield>=1.20",
+    "sgp4>=2.3",
+]
+recommended = [
+    "scipy>=1.5",
+    "matplotlib>=3.3,!=3.4.0,!=3.5.2",
+]
+all = [
+    "astropy[recommended]",  # installs the [recommended] dependencies
```

Comment:
So does this use 'recommended' from this pyproject file or does it actually get astropy from PyPI and get recommended from there?